crates/server/src/zone_handler/catalog.rs

```diff
@@ -760,6 +760,35 @@ async fn send_error_response(
     }
 }
 
+/// Strip DNSSEC records from an AuthLookup if DO bit is not set
+///
+/// As per RFC 4035 section 3.2.1, DNSSEC records should be stripped unless:
+/// - The DO bit is set, OR
+/// - The record type matches the query type (explicitly requested)
+fn maybe_strip_dnssec_records(
```

Comment:
Nit: we keep a fairly strict top-down ordering, so this should go below its caller.